### PR TITLE
[gh-198] Fix error during the generation of files on Phoenix version 1.6.7

### PR DESCRIPTION
**Pull request by byhbt**
https://github.com/nimblehq/elixir-templates/issues/198

## What happened 👀

- [x] Update the test Phoenix version in the Github workflow
- [x] Update the README content for supporting the Phoenix `1.6.11`
- [x] Remove the modification related to the `channel_case.ex`

## Insight 📝

Remove the scripts updating the `channel_case.ex`. Because of the changes in code structure when upgrading from Phoenix `1.6.6` -> `1.6.7` and later versions.

Reference diff: https://www.phoenixdiff.org/?source=1.6.6&target=1.6.11

## Proof Of Work 📹

The test should pass when using the latest Phoenix version `1.6.11`

### CODE REVIEW

File: lib/nimble_template/addons/test_env.ex [hunk 1]
```diff
@@ -133,10 +133,7 @@ defmodule NimbleTemplate.Addons.TestEnv do
   end
 
   defp edit_test_support_cases(project) do
-    project
-    |> edit_test_support_case("channel_case")
-    |> edit_test_support_case("conn_case")
-    |> edit_test_support_case("data_case")
+    edit_test_support_case(project, "data_case")
```
Comment (andyduong1920) [inline lib/nimble_template/addons/test_env.ex:136]: I guess we still have the `conn_case` on the Phoenix 1.6.11 🤔should we keep the `edit_test_support_cases(conn_case)` there @byhbt

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex [hunk 2]
```diff
@@ -27,33 +27,6 @@ defmodule NimbleTemplate.Addons.Phoenix.MixRelease do
   end
 
   defp edit_files(%{otp_app: otp_app, web_module: web_module} = project) do
-    Generator.delete_content(
-      "config/runtime.exs",
-      """
-      # Start the phoenix server if environment is set and running in a release
```
Comment (andyduong1920) [inline lib/nimble_template/addons/variants/phoenix/mix_release.ex:33]: should we keep deleting these content on Phoenix 1.6.11 @byhbt the purpose of this is delete the default config from Phoenix after generate owr project 🤔

Comment (byhbt) [reply lib/nimble_template/addons/variants/phoenix/mix_release.ex:33]: Yup, good catch, i revert the delete content here: https://github.com/nimblehq/elixir-templates/pull/250/commits/ea1c5ea6fc4df5c9561fa9badba7bb4337f3881b

However on the new version, the config content is changed. You can check here: https://www.phoenixdiff.org/?source=1.6.6&target=1.6.11 at `config/runtime.exs`

File: lib/nimble_template/addons/test_env.ex (hunk 1 above)
Comment (byhbt) [reply lib/nimble_template/addons/test_env.ex:136]: The main purpose of `edit_test_support_case` here is about clean the alias `Ecto.Adapters.SQL.Sandbox`. But on the new version, the `alias` is not used anymore. So I think we can remove the `conn_case` in this case.

They reference directly to the `datacase`:

```
setup tags do            
    SampleApp.DataCase.setup_sandbox(tags)            
                    
    {:ok, conn: Phoenix.ConnTest.build_conn()}            
end
```

Reference: https://www.phoenixdiff.org/?source=1.6.6&target=1.6.11 `test/support/conn_case.ex`

**Review by Nihisil — APPROVED** [review]

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex (hunk 2 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/mix_release.ex:33]: Got it @byhbt

File: lib/nimble_template/addons/test_env.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/test_env.ex:136]: Got it @byhbt

**Review by andyduong1920 — APPROVED** [review]

File: lib/nimble_template/addons/variants/phoenix/mix_release.ex (hunk 2 above)
Comment (andyduong1920) [reply lib/nimble_template/addons/variants/phoenix/mix_release.ex:33]: Got it @byhbt

**Pull request merged**